src/main/resources/story/walmart/SearchForProductBySearchBox.story

```diff
@@ -1,22 +1,19 @@
 Description: Story to test product search by Searchbox;
 
 Scenario: Open Walmart main page
-Given I am on a page with the URL 'https://www.walmart.com/'
+Given I am on the main application page
 Then the page title is 'Walmart.com | Save Money. Live Better.'
 
 
 Scenario: Search product by Searchbox
-When I initialize the scenario variable `searchLine` with value `Playhouse`
-When I enter '${searchLine}' in a field by the xpath '//input[@id='global-search-input']'
-When I click on an element by the xpath '//button[@id='global-search-submit']'
-!-- Check that at least one product found
-Then the number of elements found by the xpath '//div[@class='search-result-listview-item Grid']//span[contains(., '${searchLine}')] | span[contains(., #{toLowerCase("${searchLine}")})]' is greater than '0'
-
-!-- I try to read #of results found (in format " of 276 results") and store it '${numberOfResultsText}'
-When I change context to an element by the xpath '//div[@class='result-summary-container']//span[@class='visuallyhidden']'
-When I set the text found in search context to the scenario variable 'numberOfResultsText'
+When I enter '<searchLine>' in a field by the xpath '//input[@placeholder='Search']'
+When I change context to an element by the xpath '//input[@placeholder='Search']'
+When I set focus to the context element
+When I press ENTER on keyboard
 When I change context to the page
-!-- Get digist from '${numberOfResultsText}' and store it to '${numberOfResults}'
-When I initialize the scenario variable `numberOfResults` with value #{replaceFirstByRegExp(.(\d+)., $1,'${numberOfResultsText}')}
-
-!-- ToDO: check that actual #of results found equal to '${numberOfResults}'
\ No newline at end of file
+Then the text 'Sorry, no products matched "<searchLine>"' does not exist
```

Comment:
Is that check enough to be sure that at least 1 result matching `<searchLine>` is found? Or should I add addition check like in line 15, but with $comparisonRule 'greater than or equal to 1'?

Reply:
Validation for error message is a good check but its better to check that in a negative test case when you have not results found. 
But you can use it here as well, along with the check in line 15.